skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,322 @@
+_SeqData = namedtuple("SeqData", ["seq", "metadata", "pos_metadata"])
+
+
+@stockholm.sniffer()
+def _stockholm_sniffer(fh):
+    # Smells a Stockholm file if the first line contains 'Stockholm' and the
+    # version number.
+    try:
+        line = next(fh)
+    except StopIteration:
+        return False, {}
+    if line[:15] == "# STOCKHOLM 1.0":
+        return True, {}
+
+    return False, {}
+
+
+@stockholm.reader(TabularMSA)
+def _stockholm_to_tabular_msa(fh, constructor=Protein):
+    # Uses OrderedDict() to make sure dna_data isn't arranged randomly
+    dna_data = OrderedDict()
+    # Creates SeqData class to store all sequence data and make code more
+    # readable
+    metadata = {}
+    positional_metadata = {}
+    seqs = []
+
+    # Retrieves data from file, reads first so that data order will be kept
+    # consistent.
+    for line in fh:
+        if is_data_line(line):
+            dna_data = _parse_stockholm_line_data(line, dna_data)
+    fh.seek(0)
+
+    # Retrieves metadata from file
+    for line in fh:
+        if line.startswith("#=GF"):
+            metadata = _parse_stockholm_line_gf(line, metadata)
+        elif line.startswith("#=GS"):
+            dna_data = _parse_stockholm_line_gs(line, dna_data)
+        elif line.startswith("#=GR"):
+            dna_data = _parse_stockholm_line_gr(line, dna_data)
+        elif line.startswith('#=GC'):
+            positional_metadata = _parse_stockholm_line_gc(line,
+                                                           positional_metadata)
+
+    for key in dna_data.keys():
+        # Sets blank dictionaries and lists to None instead
+        # Note: _replace is not a private function, see
+        # https://docs.python.org/2/library/collections.html#namedtuple-factory
+        # -function-for-tuples-with-named-fields
+        if not dna_data[key].metadata:
+            dna_data[key] = dna_data[key]._replace(metadata=None)
+        if not dna_data[key].pos_metadata:
+            dna_data[key] = dna_data[key]._replace(pos_metadata=None)
+        # Adds each sequence to the MSA data
+        seqs.append(constructor(dna_data[key].seq,
+                                metadata=dna_data[key].metadata,
+                                positional_metadata=(dna_data
+                                                     [key].pos_metadata)))
+
+    if not seqs:
+        raise StockholmFormatError("No data present in file.")
+    if not positional_metadata:
+        positional_metadata = None
+    # Constructs TabularMSA
+    return TabularMSA(seqs, metadata=metadata,
+                      positional_metadata=positional_metadata,
+                      index=list(dna_data.keys()))
+
+
+def _parse_stockholm_line_gf(line, metadata):
+    """Takes ``#=GF`` line and returns parsed data."""
+    line = _remove_newline(line.split(' ', 2))
+    gf_feature = line[1]
+    if gf_feature in metadata.keys():
+        metadata[gf_feature] = metadata[gf_feature] + ' ' + line[2]
```

Comment:
``line[2]`` is the string associated with a GF data tag.
For example:
``#=GF DE CBS domain``
`line[1]` would be `DE` and `line[2]` would be `CBS domain`